Source/DafnyLanguageServer/Workspace/DocumentUpdater.cs

```diff
@@ -24,44 +24,48 @@ public class DocumentUpdater : IDocumentUpdater {
     }
 
     public async Task<DafnyDocument> ApplyChangesAsync(DafnyDocument oldDocument, DidChangeTextDocumentParams documentChange, CancellationToken cancellationToken) {
-      var changeProcessor = new ChangeProcessor(_logger, oldDocument, documentChange.ContentChanges, cancellationToken);
-      var mergedItem = new TextDocumentItem {
-        LanguageId = oldDocument.Text.LanguageId,
-        Uri = oldDocument.Uri,
+      var changeProcessor = new ChangeProcessor(_logger, oldDocument, documentChange.ContentChanges);
+      var mergedText = oldDocument.Text with {
         Version = documentChange.TextDocument.Version,
         Text = changeProcessor.MigrateText()
       };
-      var loadedDocument = await _documentLoader.LoadAsync(mergedItem, Verify, cancellationToken);
-      if (!loadedDocument.SymbolTable.Resolved) {
-        return new DafnyDocument(
-          loadedDocument.Text,
-          loadedDocument.Errors,
-          loadedDocument.Program,
-          changeProcessor.MigrateSymbolTable(),
-          // TODO migrate counterexamples?
-          null
-        );
+      try {
+        var newDocument = await _documentLoader.LoadAsync(mergedText, Verify, cancellationToken);
+        if (newDocument.SymbolTable.Resolved) {
+          return newDocument;
+        }
+        return MigrateDocument(mergedText, newDocument, changeProcessor, false);
```

Comment:
Is this migration so that in case of parse errors, we keep outdated resolution results since we're unable to compute new ones? Apparently the errors aren't migrated, so the new parse errors remain.

In any case, could you add a comment explaining this call?

Reply:
done